addons/mail/static/src/public/discuss_public_boot.js

```diff
@@ -61,50 +61,46 @@ export async function boot() {
     }));
     await owl.mount(MainComponentsContainer, { env, target: document.body });
 
-    async function createThreadViewFromChannelData(channelData) {
-        const messaging = await owl.Component.env.services.messaging.get();
-        const threadViewer = messaging.models['mail.thread_viewer'].create({
-            extraClass: 'flex-grow-1',
-            hasMemberList: true,
-            hasThreadView: true,
-            hasTopbar: true,
-            thread: insert(messaging.models['mail.thread'].convertData(channelData)),
-        });
-        const ThreadView = getMessagingComponent('ThreadView');
-        const threadViewComponent = new ThreadView(null, {
-            composerAttachmentsDetailsMode: 'card',
-            hasComposer: true,
-            hasComposerThreadTyping: true,
-            hasSquashCloseMessages: true,
-            threadViewLocalId: threadViewer.threadView.localId,
-        });
-        await threadViewComponent.mount(document.body);
-        if (threadViewer.thread.defaultDisplayMode === 'video_full_screen') {
-            await threadViewer.thread.toggleCall({ startWithVideo: true });
-            // TODO full screen not possible until the user actually makes an action on the page
-        }
-    }
-
-    async function createWelcomeView(channelData) {
+    async function createAndMountDiscussPublicRoot(channelData) {
         const messaging = await owl.Component.env.services.messaging.get();
         const channel = messaging.models['mail.thread'].insert(
             messaging.models['mail.thread'].convertData(channelData)
         );
-        const WelcomeView = getMessagingComponent('WelcomeView');
         const welcomeView = messaging.models['mail.welcome_view'].create({
             channel: link(channel),
             isDoFocusGuestNameInput: true,
             pendingGuestName: messaging.currentGuest && messaging.currentGuest.name,
         });
-        const welcomeViewComponent = new WelcomeView(null, {
-            localId: welcomeView.localId,
+        const threadViewer = messaging.models['mail.thread_viewer'].create({
+            extraClass: 'flex-grow-1',
+            hasMemberList: true,
+            hasThreadView: true,
+            hasTopbar: true,
+            thread: link(channel),
+        });
+        const discussPublicRoot = messaging.models['mail.discuss_public_root'].create({
+            activeView: (() => {
+                if (
+                    channel.defaultDisplayMode === 'video_full_screen' ||
+                    (messaging.currentGuest && messaging.currentGuest.name === 'Guest')
```

Comment:
This condition is too weak, I think we're better of writing something on local storage after the first visit, and only show it at the first visit.